packages/x-charts/src/ChartsVoronoiHandler/ChartsVoronoiHandler.tsx

```diff
@@ -0,0 +1,154 @@
+import { CartesianContext } from '../context/CartesianContextProvider';
+import { SVGContext, DrawingContext } from '../context/DrawingProvider';
+import { SeriesContext } from '../context/SeriesContextProvider';
+import { getValueToPositionMapper } from '../hooks/useScale';
+
+export type ChartsVoronoiHandlerProps = {
+  /**
+   * Defines the maximal distance between a scatter point and the pointer that triggers the interaction.
+   * If `undefined`, the radius is assumed to be infinite.
+   * @default undefined
+   */
+  voronoiMaxRadius?: number | undefined;
+};
+
+function ChartsVoronoiHandler(props: ChartsVoronoiHandlerProps) {
+  const { voronoiMaxRadius } = props;
+  const svgRef = React.useContext(SVGContext);
+  const { width, height, top, left } = React.useContext(DrawingContext);
+  const { xAxis, yAxis, xAxisIds, yAxisIds } = React.useContext(CartesianContext);
+  const { dispatch } = React.useContext(InteractionContext);
+
+  const { series, seriesOrder } = React.useContext(SeriesContext).scatter ?? {};
+  const voronoiRef = React.useRef<
+    Record<string, { startIndex: number; endIndex: number }> & { delauney?: Delaunay<any> }
+  >({});
+
+  const defaultXAxisId = xAxisIds[0];
+  const defaultYAxisId = yAxisIds[0];
+
+  useEnhancedEffect(() => {
+    dispatch({ type: 'updateVoronoiUsage', useVoronoiInteraction: true });
+    return () => {
+      dispatch({ type: 'updateVoronoiUsage', useVoronoiInteraction: false });
+    };
+  }, [dispatch]);
+
+  useEnhancedEffect(() => {
+    if (seriesOrder === undefined || series === undefined) {
+      // If there is no scatter chart series
+      return;
+    }
+
+    voronoiRef.current = {};
+    let points: number[] = [];
+    seriesOrder.forEach((seriesId) => {
+      const { data, xAxisKey, yAxisKey } = series[seriesId];
+
+      const xScale = xAxis[xAxisKey ?? defaultXAxisId].scale;
+      const yScale = yAxis[yAxisKey ?? defaultYAxisId].scale;
+
+      const getXPosition = getValueToPositionMapper(xScale);
+      const getYPosition = getValueToPositionMapper(yScale);
+
+      const seriesPoints = data.flatMap(({ x, y }) => [getXPosition(x), getYPosition(y)]);
+      voronoiRef.current[seriesId] = {
+        startIndex: points.length,
+        endIndex: points.length + seriesPoints.length,
+      };
+      points = points.concat(seriesPoints);
+    });
+
+    voronoiRef.current.delauney = new Delaunay(points);
+  }, [defaultXAxisId, defaultYAxisId, series, seriesOrder, xAxis, yAxis]);
+
+  React.useEffect(() => {
+    const element = svgRef.current;
+    if (element === null) {
+      return undefined;
+    }
+
+    const handleMouseOut = () => {
+      dispatch({ type: 'exitChart' });
+    };
+
+    // TODO: A perf optimisation of voronoi could be to use the last point as the intial point for the next search.
+    const handleMouseMove = (event: MouseEvent) => {
+      // Get mouse coordinate in global SVG space
+      const pt = svgRef.current!.createSVGPoint();
```

Comment:
Nit: Have you considered using a full word (i.e.: `point`, `svgPoint`) for this variable?
Or do you have an argument as to why the shorthand form is better? 🤔

Reply:
It's mostly because it's a trash variable. I create it to access the `matrixTransform` method and then it's over, So I avoid to pick a meaining full variable name

An option could be to create a helper `getSVGPoint`

```js
const svgPoint = getSVGPoint(svgRef, event)
```

Reply:
Gotcha. 👌 
Yup, the suggested helper does sound nice. 👍